src/turboServer.ts

```diff
@@ -34,9 +37,14 @@ async function startServer() {
     '/v8/artifacts/:artifactId',
     asyncHandler(async (req: any, res: any) => {
       const { artifactId } = req.params;
-      const list = await artifactApi.listArtifacts();
 
-      const existingArtifact = list.artifacts?.find(
+      if (!artifactList) {
```

Comment:
What do you think about making it a bit more robust by checking if the required artifactId exists in the cached list, if yes, skip the get request, otherwise, make the get request, cache the list, and continue.

This way, if something external uploaded a new artifact, and it requests from the server, if there is an "old" cache, it won't find the artifactId in it, therefore, it will make the GET request.

WDYT?

Reply:
good point

Reply:
Why? the first task will request artifactId `x`, the server doesn't have a cache, therefore, it will download the artifact list, and cache it, the second task, may request an artifact that is in the cached list or not, if the cached list contains it, it will not make the GET request, otherewise, it will.

Reply:
Got your point.